gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java

```diff
@@ -167,4 +169,19 @@ private static Configuration getHadoopConfiguration() {
   public static boolean isPartitioned(Table table) {
     return table.isPartitioned();
   }
+
+  /**
+   * @param fs User configured filesystem of the target table
+   * @param userSpecifiedPath user specified path of the copy table location or partition
+   * @param existingTablePath path of an already registered Hive table or partition
+   * @return true if the filesystem resolves them to be equivalent, false otherwise
+   */
+  public static boolean areTablePathsEquivalent(FileSystem fs, Path userSpecifiedPath, Path existingTablePath) throws IOException {
+    try {
+      return fs.resolvePath(existingTablePath).equals(fs.resolvePath(userSpecifiedPath));
```

Comment:
Yes in the scenario where there are mixed filesystems, the only passing case will be when the user specifies a virtual filesystem instead of the physical one. But once that is done then it should pass as expected due to `resolvePath()`